order/layout.go

```diff
@@ -0,0 +1,66 @@
+package order
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/brimdata/zed/field"
+)
+
+var Nil = Layout{}
```

Comment:
Nit: I don't think `Nil` is a great name for this because it's a zero value rather than a `nil` value. Maybe call it `Unset`?

Reply:
I was following the ksuid pattern which has ksuid.Nil because you can't use `nil` with this type just like Layout. Seemed like a good idea.   I wasn't bother by the idea that nil is the zero value for pointers, interfaces, etc in Go but package.Nil is the zero value of the thing package implements.  You would never have a need to define Nil if you could use nil instead so that seems really clear to me.